src/Client/ClickHouseClient.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimPod\ClickHouseClient\Client;
+
+use SimPod\ClickHouseClient\Format\Format;
+use SimPod\ClickHouseClient\Output\Output;
+
+interface ClickHouseClient
+{
+    public function executeQuery(string $sql) : void;
```

Comment:
možná bych nechal jen execute?

Reply:
execute what ale?

Reply:
client.execute cokoliv, treba psr request

Reply:
to nechápu ... plánuješ tam mít takovýhle metody? K čemu by to bylo? Je to jen CH client, takže jiný věci než SQL tam nebudou, ne?

Reply:
s jistotou bych si tak zadelaval na bc break :D

Reply:
treba z toho v budoucnu vyleze, ze dava smysl rozdelit read a write executes

mrkni kolik toho exekutujou tady https://docs.oracle.com/javase/7/docs/api/java/sql/Statement.html

Reply:
ok, tak proč select nemáš podobně? :P

Reply:
jak to myslis podobne? podobne cemu?

Reply:
execute ... taky můžeš selectovat na základě více vstupů, pokud můžeš mít execute pro více vstupů ... protože rozdíl mezi execute a select je jen ten, že execute zahazuje výsledek (nebo ho ani nedostane)

Reply:
no select uz je metoda pro konkretni case. stejne jako insert

to executeQuery je obecna metoda na execute query

ja moc nerozumim otazce

Reply:
otázka je pořád stejná: proč ten suffix query. Tvrdíš, že je třeba, protože jde query předat " ve více formátech". Tak pak mě zajímá, proč u select to není issue, když to příjmá úplně stejné query, a tedy by pro to měli platit úplně stejný podmínky.

Reply:
ja rikam ze execute je moc obecny, protoze muzu execute cokoliv. execute query jasne specifikuje intent

select taky jasne specifikuje intent

Reply:
... ale co chceš executovat jinýho? :D pokud psr request, jak jsi zmínil, tak proč bys nemohl dělat selectPsrRequest?

Reply:
protoze selectovat psr request nejde. select je v contextu CH dostatecne unikatni

Reply:
Proč by nešlo? :D Nechápu jaký je rozdíl mezi execute a select podle tebe ... jsou to ty samé věci přece, jen jedna nemá output

Reply:
protoze to nejde :D veta "select request" nedava smysl. jak jsem psal, select je v terminologii jednoznacny, execute je nic a vsechno